test/controllers/ControllerBaseSpec.scala

```diff
@@ -20,29 +20,47 @@ import akka.actor.ActorSystem
 import akka.stream.{ActorMaterializer, Materializer}
 import common.SessionKeys
 import config.{AppConfig, ServiceErrorHandler}
+import controllers.predicates.{AgentPredicate, HybridUserPredicate}
 import mocks.MockAppConfig
 import org.scalamock.scalatest.MockFactory
 import org.scalatest.BeforeAndAfterEach
 import org.scalatestplus.play.guice.GuiceOneAppPerSuite
-import play.api.i18n.MessagesApi
+import play.api.i18n.{Lang, Messages, MessagesApi}
 import play.api.inject.Injector
-import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded}
+import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded, MessagesControllerComponents}
 import play.api.test.FakeRequest
-import play.filters.csrf.CSRF.Token
-import play.filters.csrf.{CSRFConfigProvider, CSRFFilter}
+import services.{AccountDetailsService, EnrolmentsAuthService}
+import uk.gov.hmrc.auth.core.AuthConnector
 import uk.gov.hmrc.http.{SessionKeys => GovUKSessionKeys}
 import uk.gov.hmrc.play.test.UnitSpec
+import views.html.errors.{AgentUnauthorised, Unauthorised}
+
+import scala.concurrent.ExecutionContext
 
 class ControllerBaseSpec extends UnitSpec with MockFactory with GuiceOneAppPerSuite with BeforeAndAfterEach {
 
   lazy val injector: Injector = app.injector
-  lazy val messages: MessagesApi = injector.instanceOf[MessagesApi]
+  lazy val messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
+  implicit lazy val messages: Messages = messagesApi.preferred(Seq(Lang("en")))
   lazy val mockServiceErrorHandler: ServiceErrorHandler = injector.instanceOf[ServiceErrorHandler]
+  val mcc: MessagesControllerComponents = injector.instanceOf[MessagesControllerComponents]
+
+  implicit val ec: ExecutionContext = injector.instanceOf[ExecutionContext]
   implicit val mockAppConfig: AppConfig = new MockAppConfig(app.configuration)
   implicit val system: ActorSystem = ActorSystem()
   implicit val materializer: Materializer = ActorMaterializer()
   implicit lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest()
 
+  val mockAgentUnauthorised: AgentUnauthorised = injector.instanceOf[AgentUnauthorised]
```

Comment:
Could you rename a few of these? Specifically if something isn't a mock (i.e. isn't defined by `mock[X]`) could you remove the word "mock" from the val name.

Not sure if you are aware but if you right click the val and choose Refactor > Rename then it will rename all of its usages too just to avoid you doing it manually.